lib/core/errors.js

```diff
@@ -195,6 +195,15 @@ class RequestRetryError extends UndiciError {
   }
 }
 
+class NGHTTP2InternalError extends UndiciError {
```

Comment:
Let's scope it to only `HTTP2`, so maybe we can do something like `HTTP2Error`?
So we can reuse it in other scenarios that might need similar approach